src/kudu/consensus/consensus_queue.cc

```diff
@@ -692,7 +730,21 @@ Status PeerMessageQueue::RequestForPeer(const string& uuid,
     unreachable_time = MonoTime::Now() - peer_copy.last_communication_time;
 
     RETURN_NOT_OK(routing_table_container_->NextHop(
-          local_peer_pb_.permanent_uuid(), uuid, &next_hop_uuid));
```

Comment:
where we making 2 calls into the routing table before this?

Reply:
Yes. Once here and the other one in `Peer::SendNextRequest()` when it calls `GetNextRoutingHopFromLeader()`. This PR removes the `GetNextRoutingHopFromLeader()` call in `Peer::SendNextRequest()`